dashboard/lib/tasks/asset_sync.rake

```diff
@@ -0,0 +1,11 @@
+namespace :assets do
+  desc 'Synchronize assets to S3'
+  task :sync => :environment do
+    # AWS CLI implements an optimized `sync` utility without any Ruby SDK equivalent.
+    `aws s3 sync #{CDO.dashboard_dir}/public/assets s3://#{CDO.assets_bucket}/#{rack_env}/assets --acl public-read --cache-control 'max-age=31536000'`
+  end
+end
+
+Rake::Task['assets:precompile'].enhance do
+  Rake::Task['assets:sync'].invoke if CDO.daemon
```

Comment:
[It looks like `daemon` is true in development too](https://github.com/code-dot-org/code-dot-org/blob/staging/deployment.rb#L123) (but not adhoc, I guess?).  If so, I think that means dev boxes will try to push assets to s3://cdo-dist/development, but they'll be retrieving assets from the local machine (because they don't run the varnish or cloudfront stuff).  Is that correct?